### DISCUSSION TITLE
How can I help

### DISCUSSION THREAD

**Discussion by stevemandl** [cdmoro/bootstrap-vue-3#1252] [Q&A] [answered]
I'm using this library in a project and keep running across things that are missing from the implementation or documentation. I can keep submitting issues as I find things if that is the best way to help, but I may also be able to submit PRs. I would prefer to do so with some guidance though, since I don't want to spend the time if my PRs are never going to be used.

I guess I could use some help with coding first - I would want to know if anyone else is working on something before I started so I could make sure I'm not creating conflicts or duplicating effort. Then I'd also like to know if I can contribute to the docs - If I end up changing the way something works, there ought to be an example in the docs of how to use it correctly.

**Comment by VividLemon** [comment, marked as answer]
In an issue thread https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1120 , you can see that the person proceeded to ask many questions on how to help and what we could do to proceed. Asking questions is encouraged. You're free to ask anything that would help you submit a PR. The library is basically just like any Vue application, so developing is not all too dissimilar from just building an app. The docs are built using Vitepress, which is basically just MD files that can use Vue. As long as you follow the established style, there's not much to say about it. The things listed on the https://github.com/orgs/bootstrap-vue-next/projects/1 are fairly up to date, but those issues likely require a bit more technical knowledge about Vue, Bootstrap and the library. In terms of just adding missing stuff, depending on the item, it may be best to create an issue and wait for the "stale exempt", as some things may be removed. Though, this doesn't happen often. I'm just thinking about someone saying "I saw a missing directive", which most from Bootstrap-vue, I have chosen to remove. (Bootstrap-vue isn't a utility library!) However, typically things like missing or malfunctioning props are usually safe additions. I think it's unlikely that you will find an issue that someone else is working on. Perhaps if the activity of the library was more active we'd require "assignees" for issues, but there are a handful of people that submit PRs, and a significantly lower portion of people that submit PRs often.

**Reply by stevemandl** [reply]
Sounds good. I will use issues to continue the questions.

**Comment by anrolmar** [comment]
Hi, I'm a Senior Vue Developer and I'd want to know how I could help on the development of this library.

I used the version for Vue 2 and I'm using the last version (for Vue3) on one of my projects.

Thanks in advance

**Reply by VividLemon** [reply]
Most things on the issues list are able to be fixed. However, many of the things on the list are advanced issues that could require a fair bit of knowledge of the system. https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues/1658 seems like an easy fix. Adding missing features from BV is usually pretty straightforward. You'd just have to go through the BV documentation and compare what they have declared as props, and then compare it to the component definitions here (not our docs, they can usually be outdated). In addition to that, a robust list of all missing features (or features that just don't work, or props that are unused) is something that could be an immense help, although getting the parity list of that is likely to be super tedious, yet required

**Reply by anrolmar** [reply]
Ok, I'll take a look into the list.

How do you work with code? I suppose I need to do a branch in my local environment and once I have finish a ticket, I should do a PR, right? To master?

**Reply by anrolmar** [reply]
Another question. How could I assign to me an issue?

**Reply by anrolmar** [reply]
How could I know if an issue is being fixed by someone?

**Reply by VividLemon** [reply]
> Another question. How could I assign to me an issue?

Github has no functionality for allowing those without write access to be assigned to an issue. If you want to start fixing something, I suppose just commenting on the issue that you're working on it is fine

> How could I know if an issue is being fixed by someone?

Assume it's not. I assign various issues to myself, but other than that the library has low enough traffic that spontaneously fixing an issue will likely have no conflicts.

> Is this one, https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues?

Yes

**Comment by anrolmar** [comment]
Where is the issues list?

**Reply by anrolmar** [reply]
Is this one, https://github.com/bootstrap-vue-next/bootstrap-vue-next/issues?
